### PR TITLE
fix: add rating selector and validation

### PR DESCRIPTION
**Pull request by WojtekTheWebDev**
Add review form was missing the selection of stars.

## Description

SfRating cannot be used as an input (yet) but there is a SfSelect component to pick the rating and SfRating to show it as stars.

## Screenshots (if appropriate):
Screenshot 2021-12-08 at 13 50 06

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by darocha — APPROVED** [review]
Nicely done!

**Review by dhargitai — APPROVED** [review]
Looks good 👍

File: packages/theme/components/AddReview.vue [hunk 1]
```diff
@@ -13,28 +13,41 @@
...
       <div v-if="isFormVisible" class="add_review--form">
-          <form @submit="submit">
+          <form class="form" @submit="submit">
               <SfInput
                 v-model="title"
                 class="form__element"
                 name="title"
                 label="Review title"
                 :required="true"
+                :valid="titleBlur || validText(title)"
```
Comment (zoltantakacs-born) [inline packages/theme/components/AddReview.vue:23]: is it always valid when the input is blurred?

File: packages/theme/components/AddReview.vue [hunk 2]
```diff
@@ -13,28 +13,41 @@
...
               <SfTextarea
                   v-model="message"
                   class="form__element"
+                  label="Review message"
                   name="message"
                   placeholder="type review message"
                   cols="50"
                   rows="5"
                   :required="true"
+                  :valid="messageBlur || validText(message)"
```
Comment (zoltantakacs-born) [inline packages/theme/components/AddReview.vue:49]: same as with the title

File: packages/theme/components/AddReview.vue [hunk 3]
```diff
@@ -65,21 +78,36 @@ export default defineComponent({
...
 
+    const validText = (text) => {
+      return text.length > 3;
+    };
+
     // eslint-disable-next-line no-undef
     const submit = (e: Event) => {
       e.preventDefault();
+
+      if (!(validText(title.value) || validText(message.value))) {
```
Comment (zoltantakacs-born) [inline packages/theme/components/AddReview.vue:97]: !validText(message.value)

File: packages/theme/components/AddReview.vue [hunk 4]
```diff
@@ -94,23 +122,32 @@ export default defineComponent({
...
 
 <style lang="scss" scoped>
 @import "~@storefront-ui/vue/styles";
-
 .form {
+  .sf-rating {
+    padding: 0 0 var(--spacer-sm) 0;
+    .sf-rating__icon {
+      --icon-size: var(--font-size--lg) !important;
+    }
```
Comment (zoltantakacs-born) [inline packages/theme/components/AddReview.vue:149]:
```
&__icon {
...
}
```
---
is important inevitable?

**Review by zoltantakacs-born — CHANGES_REQUESTED** [review]

File: packages/theme/components/AddReview.vue (hunk 1 above)
Comment (WojtekTheWebDev) [reply packages/theme/components/AddReview.vue:23]: I used similar logic like SfUI team in their form example 
Screenshot 2021-12-09 at 09 49 23
It's just for the ui purposes to not tigger the error when the focus is on the input

File: packages/theme/components/AddReview.vue (hunk 2 above)
Comment (WojtekTheWebDev) [reply packages/theme/components/AddReview.vue:49]: The same response as in title
https://github.com/vuestorefront/bigcommerce/pull/87#discussion_r765558434

**Review by zoltantakacs-born — APPROVED** [review]

**Pull request merged**